### DISCUSSION TITLE
[Feature Request] 有考虑在文档加入 codesandbox和stackblitz 交互支持吗？

### DISCUSSION THREAD

**Discussion by LadyChatterleyLover** [youzan/vant#12589] [General]
### 这个功能解决了什么问题？

现在大部分组件库文档都支持在线交互，如果有计划加入这个功能，我可以提供pr。

### 你期望的 API 是什么样子的？

加入codesandbox和stackblitz在线交互

**Comment by LadyChatterleyLover** [comment]
我做了一个在线的playground,  https://vant-playground.vercel.app/   如果合适的话可以加在项目里面

**Comment by chenjiahan** [comment]
> 有考虑在文档加入 codesandbox和stackblitz 交互支持吗

我比较好奇怎么维护 codesandbox 相关的各种示例代码，你能介绍一下吗

**Comment by chenjiahan** [comment]
> 我做了一个在线的playground, https://vant-playground.vercel.app/ 如果合适的话可以加在项目里面

很棒！我会加到 Vant 的 Community Ecosystem 里面，请问有对应的 GitHub 仓库地址码

**Comment by LadyChatterleyLover** [comment]
> > 有考虑在文档加入 codesandbox和stackblitz 交互支持吗
> 
> 我比较好奇怎么维护 codesandbox 相关的各种示例代码，你能介绍一下吗

需要将文档示例中的源代码传给平台，平台有对应的api可以解析，构建一个简单的项目来跑demo，不过就需要做一个插件来解析文档的源代码

**Comment by chenjiahan** [comment]
现在的 demo 基本上是一个或多个 Vue 组件，如果要拆分为可以在 codesandbox 的代码单元，感觉会需要对 demo 代码进行一定的结构化调整？

**Comment by LadyChatterleyLover** [comment]
> > 我做了一个在线的playground, https://vant-playground.vercel.app/ 如果合适的话可以加在项目里面
> 
> 很棒！我会加到 Vant 的 Community Ecosystem 里面，请问有对应的 GitHub 仓库地址码

https://github.com/LadyChatterleyLover/vant-playground   目前放在我自己仓库中的哈

**Comment by LadyChatterleyLover** [comment]
> 现在的 demo 基本上是一个或多个 Vue 组件，如果要拆分为可以在 codesandbox 的代码单元，感觉会需要对 demo 代码进行一定的结构化调整？

是的，需要对demo代码进行调整，做成一个组件或者插件形式，必须要获取到示例的源代码才行，

**Comment by LadyChatterleyLover** [comment]
目前最快的方式是基于vite的 import xxx from 'xxx?raw'获取到文件的源代码

**Comment by chenjiahan** [comment]
如果改造成本不是太高的话，我觉得加入 codesandbox 支持还是非常有帮助的，欢迎提一个最小 PR 来预览一下效果~

另外，相较于 stackblitz 来说，我更倾向于优先支持 codesandbox，stackblitz 可以等以后再考虑，这两者的功能其实是相似的。

**Comment by LadyChatterleyLover** [comment]
好的，空了我提一个，



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: neverland ***@***.***&gt;
发送时间: 2023年9月28日 20:11
收件人: youzan/vant ***@***.***&gt;
抄送: luopei ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [youzan/vant] [Feature Request] 有考虑在文档加入 codesandbox和stackblitz 交互支持吗？ (Issue #12325)





 
如果改造成本不是太高的话，我觉得加入 codesandbox 支持还是非常有帮助的，欢迎提一个最小 PR 来预览一下效果~
 
另外，相较于 stackblitz 来说，我更倾向于优先支持 codesandbox，stackblitz 可以等以后再考虑，这两者的功能其实是相似的。
